streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/KTableKTableJoinNode.java

```diff
@@ -120,30 +115,13 @@ private void enableVersionedSemantics(final ProcessorParameters<K, ?, ?, ?> proc
     public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
         final String thisProcessorName = thisProcessorParameters().processorName();
         final String otherProcessorName = otherProcessorParameters().processorName();
-        final String mergeProcessorName = mergeProcessorParameters().processorName();
 
-        topologyBuilder.addProcessor(
-            thisProcessorName,
-            thisProcessorParameters().processorSupplier(),
-            thisJoinSideNodeName());
-
-        topologyBuilder.addProcessor(
-            otherProcessorName,
-            otherProcessorParameters().processorSupplier(),
-            otherJoinSideNodeName());
-
-        topologyBuilder.addProcessor(
-            mergeProcessorName,
-            mergeProcessorParameters().processorSupplier(),
-            thisProcessorName,
-            otherProcessorName);
+        thisProcessorParameters().addProcessorTo(topologyBuilder, thisJoinSideNodeName());
+        otherProcessorParameters().addProcessorTo(topologyBuilder, otherJoinSideNodeName());
+        mergeProcessorParameters().addProcessorTo(topologyBuilder, thisProcessorName, otherProcessorName);
 
         topologyBuilder.connectProcessorAndStateStores(thisProcessorName, joinOtherStoreNames);
         topologyBuilder.connectProcessorAndStateStores(otherProcessorName, joinThisStoreNames);
```

Comment:
we should be able to remove these. I would actually consider it a bug if we were to remove these and the tests didn't pass -- since #addProcessorTo should call #addStateStore which should call #connectProcessorAndStateStores

Reply:
Ah I had missed this but this is necessary because it's connecting `this` with `other` and vice versa. I don't think `addProcessorTo` can do that by default, right?

Reply:
let me double check on this but my understanding is that if both stores are used by both "thisProcessor" and "otherProcessor", and vice versa, then we should have the `#stores` method return both  stores.

Reply:
Following up on this -- make sure to read my comment [here](https://github.com/apache/kafka/pull/18048/files#r1879285311) first for full context -- the reason we have to explicitly connect these stores with the opposite side's processor and can't do this via the `#stores` implementation is because the stores are actually associated with the upstream KTable, and not with any of the join nodes. So each KTable of course only returns its own state store from its own `#stores` method (what you implemented in your last KIP-1112 PR), since that processor only needs access to its own store. And the downstream joins just connect to the upstream KTable's store

So this explains why removing these `#connectProcessorAndStateStores` makes things fail. One question I still have is whether we should try to get a handle on the upstream KTable's store itself and return that from the `#stores` method of the downstream join that uses that store, rather than have them be connected manually using `#connectProcessorAndStateStores`. At the moment I'm thinking (and hoping) that it's ok to leave things as-is, since the join nodes actually don't access the stores directly but instead go through something called a `ValueGetter`. But I'm planning to tackle table-table joins next in my end-to-end integration tests of processor/store wrapping, so we'll see if this becomes an issue